vcpkg.json

```diff
@@ -0,0 +1,7 @@
+{
+    "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/main/docs/vcpkg.schema.json",
+    "builtin-baseline": "b02e341c927f16d991edbd915d8ea43eac52096c",
+    "dependencies": [
+      "zlib"
```

Comment:
When building right now, I still receive cmake errors. How can we prevent this?

```
CMake Error at C:/Program Files (x86)/cmake/cmake-3.30.4-windows-x86_64/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find ZLIB (missing: ZLIB_LIBRARY ZLIB_INCLUDE_DIR)
Call Stack (most recent call first):
  C:/Program Files (x86)/cmake/cmake-3.30.4-windows-x86_64/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files (x86)/cmake/cmake-3.30.4-windows-x86_64/share/cmake-3.30/Modules/FindZLIB.cmake:202 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  D:/Projects/Microsoft/vcpkg/scripts/buildsystems/vcpkg.cmake:893 (_find_package)
  Core/Libraries/Source/Compression/CMakeLists.txt:42 (find_package)
```

Reply:
No it stops with
```
Configuring incomplete, errors occurred!
```

Reply:
I would try implement it like so

```
if (windows || not has vcpkg)
  use fetched zlib
else
  use vcpkg zlib
```

This way windows builds are not changed.